RavenDB-15108

aviv86 opened the pull request:
https://issues.hibernatingrhinos.com/issue/RavenDB-15108

aviv86 committed:
RavenDB-15108 : TimeSeriesStats - introduce 'Name' Column which will hold the series name in it's original casing.
implement GetTimeSeriesNameOriginalCasing, GetTimeSeriesNamesForDocumentOriginalCasing

aviv86 committed:
RavenDB-15108 : add Name (original casing) to CounterReplicationItem, and use it in IncomingReplicationHandler instead of lower-name

aviv86 committed:
RavenDB-15108 : TimeSeriesStorage
- use original name (from stats) when creating CounterReplicationItem
- replace GetTimeSeriesLowerNamesForDocument by GetTimeSeriesNamesForDocument, remove unused code
- AddTimeSeriesNameToMetadata() - do not throw on conflict

aviv86 committed:
RavenDB-15108 : DocumentPutAction
- RecreateTimeSeries  : GetMetadata should use the new GetTimeSeriesNamesForDocument
- remove IsReturningLowerCasedMetadata from IRecreationType
- removed unused code

aviv86 committed:
RavenDB-15108 : add tests, DeleteConflictsFor() - if the conflicted doc has time-series flag, add 'ResolveTimeSeriesConflict' to non-persistent-flags

karmeli87 commented on src/Raven.Server/Documents/TimeSeries/TimeSeriesStats.cs:
```diff
@@ -142,13 +148,20 @@ public void UpdateStats(DocumentsOperationContext context, TimeSeriesSliceHolder
+            }
+            else
+            {
+                nameScope = Slice.From(context.Allocator, slicer.Name, out nameSlice);
```
Let the slicer hold also the original nameSlice

aviv86 replied:
done

karmeli87 commented on src/Raven.Server/Documents/TimeSeries/TimeSeriesStats.cs:
```diff
@@ -268,6 +283,39 @@ private static DateTime GetLastLiveTimestamp(DocumentsOperationContext context,
+            }
+        }
+
+        public LazyStringValue GetTimeSeriesNameOriginalCasing(DocumentsOperationContext context, Slice storageKey)
```
do we need all of the different `GetOriginalName` if we have this method?

karmeli87 commented on test/SlowTests/Client/TimeSeries/TimeSeriesAndDocumentConflicts.cs:
```diff
@@ -0,0 +1,393 @@
+using Xunit;
+using Xunit.Abstractions;
+
+namespace SlowTests.Client.TimeSeries
```
What if we have a conflict on a time-series casing?
e.g. "Heartrate" vs "HeartRate"

aviv86 replied:
handled

aviv86 committed:
RavenDB-15108 : address PR comments:
- add NameSlice to SliceHolder
- Incoming TS replication - if we have a conflict on a time-series casing, choose the name with lower lexical value,
and update the name in local Stats and in document's metadata

karmeli87 reviewed (approved):
LGTM
